Client/src/Pages/Authentication/login.js

```diff
@@ -33,139 +50,151 @@ function SignIn({t, i18n}) {
         sessionStorage.setItem("id", json.preferred_username);
         cookies.set("role", "Student", { path: "/" });
-        fetchPicture()
-      }
-    }catch(error){
-      if(error.code =="NotAuthorizedException"){
-        setValidInfo(false);   
+        fetchPicture();
       }
-      else{
-        alert(error.message)
+    } catch (error) {
+      if (error.code == "NotAuthorizedException") {
+        setValidInfo(false);
+      } else {
+        alert(error.message);
       }
       console.log(error);
     }
-  }
+  };
 
   //checks if they put in an email and password
   function validateForm() {
     return email.length > 0 && password.length > 0;
   }
 
-  //So the user can press enter rather than click the button
-  function handleKeyPress(event){
-    if(event.key === 'Enter' && validateForm()){
-      return handleSubmit
-    }
-  }
+  //hides and shows password
+  const togglePasswordVisiblity = () => {
+    setPasswordVisibility(!isPasswordVisible);
+  };
 
   //Notably here the button is disabled if the form isn't validated
   return (
     <Container component="main" maxWidth="xs">
       <div className={classes.paper}>
-        <img  height="175" width="175" src ="cicscareers_logo_3.png"></img>
-        <br/>
+        <img height="175" width="175" src="cicscareers_logo_3.png"></img>
+        <br />
+
         <form className={classes.form}>
           <TextField
             variant="outlined"
             margin="normal"
             required
             fullWidth
             id="email"
-            label={t('email')}
+            label={t("email")}
             name="email"
             autoComplete="email"
             autoFocus
-            form className={classes.form}
-            onChange={e => setEmail(e.target.value)}
+            form
+            className={classes.form}
+            onChange={(e) => setEmail(e.target.value)}
           />
           <TextField
             variant="outlined"
             margin="normal"
             required
             fullWidth
             name="password"
-            label={t('password')}
-            type="password"
+            label={t("password")}
+            type={isPasswordVisible ? "text" : "password"}
             id="password"
-            form className={classes.form}
+            form
+            className={classes.form}
             autoComplete="current-password"
-            onChange={e => setPassword(e.target.value)}
-            onKeyPress={handleKeyPress}
+            onChange={(e) => setPassword(e.target.value)}
           />
+          <InputAdornment className={classes.eyeIcon}>
+            <IconButton onClick={togglePasswordVisiblity}>
+              {isPasswordVisible ? (
+                <VisibilityOffIcon />
+              ) : (
+                //onClick so the user doesn't accidentally trigger it
+                <VisibilityIcon />
+              )}
+            </IconButton>
+          </InputAdornment>
```

Comment:
Sorry I wasn't super clear. You can actually put the input adornment as a "prop" of the input, so then you don't need to do anything with CSS at all! And then it will show up inside the textbox rather than floating on the screen. Like this. It will require that TextField becomes OutlinedInput though (you can just import this from material ui)
```
<OutlinedInput
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="password"
            label={t("password")}
            type={isPasswordVisible ? "text" : "password"}
            id="password"
            className={classes.form}
            autoComplete="current-password"
            onChange={(e) => setPassword(e.target.value)}
            endAdornment={
              <InputAdornment position="end">
                <IconButton onClick={togglePasswordVisiblity}>
                  {isPasswordVisible ? (
                    <VisibilityOffIcon />
                  ) : (
                    //onClick so the user doesn't accidentally trigger it
                    <VisibilityIcon />
                  )}
                </IconButton>
              </InputAdornment>}
          />
```

Reply:
I made the changes you asked for. Please take a look.